### ISSUE TITLE
Library "Non-normalized Paths" misses SourceLink document paths, and "paths normalized" reports a false clean

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3410]
The library `Non-normalized Paths` section reports **only** the CodeView PDB path. It never reports non-normalized SourceLink `documents` keys, even though the sibling `AssemblyInspector` path does exactly that and the data is already in hand.

The consequence is worse than a missing row: the same code sets `HasNormalizedPaths = true` on PDB-path evidence alone, so an assembly with a deterministic `/_/` PDB path but non-normalized SourceLink document keys is reported as **clean**.

## Evidence

The library command has exactly one source for this data:

| Location | Behavior |
| --- | --- |
| `src/dotnet-inspect/Inspectors/LibraryMetadataService.cs:130` | `inspection.NonNormalizedPaths = pdbContext.NonNormalizedPaths;` — the only assignment |
| `src/ILInspector.Metadata/PdbContext.cs:1073` | the only writer: `NonNormalizedPaths.Add($"PDB Path: {cvData.Path}")` |

`grep -c "SourceLink:" src/ILInspector.Metadata/PdbContext.cs` returns **0** — `PdbContext` cannot emit a SourceLink row under any input.

`PdbContext` already extracts the SourceLink JSON it would need, at lines 322 and 1108:

```csharp
SourceLinkJson = AssemblyInspector.ExtractSourceLinkFromReader(_pdbReader);
```

It simply never calls `CheckSourceLinkPaths`.

By contrast `src/ILInspector.Metadata/AssemblyInspector.cs` covers both axes:

| Line | Row emitted |
| --- | --- |
| 227 | `PDB Path: {cvData.Path}` |
| 248-255 | `SourceLink: {path}` for every `documents` key not starting with `/_`, via `CheckSourceLinkPaths` (line 400) |

So the two paths disagree about what "non-normalized" means, and the library command — the one users reach through `-S "Non-normalized Paths"` — has the weaker definition.

## Impact

- A `SourceLink:` row is unreachable from the library command.
- `HasNormalizedPaths` / the `Signals` "paths normalized" answer inherit the blind spot and report a false clean.
- The section name promises path normalization generally, but delivers one of the two sources.

## Observed today (PDB-path axis works)

```
$ dotnet-inspect library log4net -S "Non-normalized Paths"
## Non-normalized Paths
- PDB Path: /home/alle/git/apache/logging-log4net/src/log4net/obj/Release/netstandard2.0/log4net.pdb

$ dotnet-inspect library RestSharp -S "Non-normalized Paths"
## Non-normalized Paths
- PDB Path: /home/runner/work/RestSharp/RestSharp/src/RestSharp/obj/Release/net10.0/RestSharp.pdb
```

Both are detected via the CodeView path. Neither exercises the SourceLink axis.

## Proposed fix

Call `CheckSourceLinkPaths` from `PdbContext` where `SourceLinkJson` is populated, emitting `SourceLink: <key>` rows and folding the result into `HasNormalizedPaths` — matching `AssemblyInspector` so the two producers agree.

Since the shared rule already exists in `AssemblyInspector.CheckSourceLinkPaths`, prefer reusing it over adding a second normalization predicate.

## Evidence required

- A fixture or real package whose PDB path is normalized (`/_/`) but whose SourceLink `documents` keys are not — this is the case that currently reports a false clean and must flip.
- A close negative: a fully normalized assembly must stay silent (no new rows, `HasNormalizedPaths` still true).
- Confirm the existing PDB-path rows (log4net, RestSharp above) are unchanged.
